src/main/java/com/GG/T9AgeCombat/service/repository/UnitService.java

```diff
@@ -1,21 +1,152 @@
 package com.GG.T9AgeCombat.service.repository;
```

Comment:
We don't have to do this now necessarily but it probably makes sense to extend the `getSpecialRuleByUnitId` to also get the unit's special rules based on their height classification.